synapse/storage/databases/main/roommember.py

```diff
@@ -504,6 +504,63 @@ def _get_rooms_for_user_with_stream_ordering_txn(
             for room_id, instance, stream_id in txn
         )
 
+    @cachedList(
+        cached_method_name="get_rooms_for_user_with_stream_ordering",
+        list_name="user_ids",
+    )
+    async def get_rooms_for_users_with_stream_ordering(
```

Comment:
may want a little docstring here, mostly for the return type, which is not very obvious.

It's roughly `user → (room, event_pos)` to my reading. What does the `event_pos` mean here?

… seems to be the position of the event that gives the user their membership in the room?

Reply:
Do we have no need to invalidate this if a user leaves or so on?

Reply:
Ah, missed that this didn't have a docstring.

I think you're missing that this is using a `@cachedList`, which means that its simply a batch version of `get_rooms_for_user_with_stream_ordering` and reuses the same cache. Hence why it includes the stream ordering and the like. Also means that invalidation is already taken care of.

Reply:
Oooh, I see, very cute. That makes sense!